Trappist/src/Promact.Trappist.Core/Controllers/CategoryController.cs

```diff
@@ -26,44 +30,47 @@ public CategoryController(ICategoryRepository categoryRepository)
 
         [HttpPost]
         /// <summary>
-        /// Post Method 
-        /// Will Add a Catagory Name in Category Model
+        /// API to add Category
         ///</summary>
-        /// <param name="category">Object of  class Category</param>
-        /// <returns>object of the class </returns>
-        public IActionResult CatagoryAdd([FromBody] Category category)
+        /// <param name="category">Category object</param>
+        /// <returns>Category object</returns>
+        public async Task<IActionResult> AddCategory([FromBody] Category category)
         {
             if (!ModelState.IsValid)
             {
                 return BadRequest(ModelState);
             }
-            _categoryRepository.AddCategory(category);
+            if (await _categoryRepository.IsCategoryNameExistsAsync(category))
```

Comment:
IsCategoryNameExistsAsync Method need both id and category name as parameter 
so I take entire category Object